modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl

```diff
@@ -230,15 +230,20 @@ public abstract class Base${schemaName}ResourceImpl
 							for (${schemaName} ${schemaVarName} : ${schemaVarNames}Page.getItems()) {
 								${schemaVarName}.setPermissions(
 									() -> NestedFieldsSupplier.supply("permissions", nestedField -> {
-										Page<Permission> permissionsPage = get${schemaName}PermissionsPage(
-											<#if properties?keys?seq_contains("id")>
-												${schemaVarName}.getId()
-											<#elseif properties?keys?seq_contains(schemaVarName + "Id")>
-												${schemaVarName}.get${schemaVarName}Id()
-											<#else>
-												${schemaVarName}Id
-											</#if>
-
+										Page<Permission> permissionsPage =
+										<#if freeMarkerTool.getPermissionsPageJavaMethodSignature("get", javaMethodSignatures, schemaName)??>
+											get${schemaName}PermissionsPage(
+												<#if properties?keys?seq_contains("id")>
+													${schemaVarName}.getId()
+												<#elseif properties?keys?seq_contains(schemaVarName + "Id")>
+													${schemaVarName}.get${schemaVarName}Id()
+												<#else>
+													${schemaVarName}Id
+												</#if>
+										<#elseif freeMarkerTool.getParentPermissionsPageJavaMethodSignature("get", javaMethodSignatures, parentSchemaName, schemaName)??>
+											get${parentSchemaName}${schemaName}PermissionsPage(
+												${parentSchemaName?uncap_first}ExternalReferenceCode, ${schemaVarName}.getExternalReferenceCode()
```

Comment:
I think this does not follow the current pattern. I think it should be in the same line